requirements.txt

```diff
@@ -14,6 +14,7 @@ pandas == 1.3.5
 # TestCompactionSession::test_compact_partition_when_incremental_then_rcf_stats_accurate to fail
 # due to input_inflation exceeding 1e-5
 pyarrow == 17.0.0
+pyarrow-stubs==17.0.0.0
```

Comment:
Do we need this?

Reply:
@pdames - pyarrow lib itself doesn't expose any type information, which makes the dev work super annoying and voids all the compile time check benefits. Adding the stubs helps surfacing a user-friendly python interface for pyarrow.

Please let me know if this addresses your concern.

Reply:
Hmm... I think this comes down to whether or not your IDE supports, or can be configured to support, autocomplete against Cython .pyx, .pxd, and .pxi files (I haven't bumped into this issue with PyCharm). If you require it for your dev setup, I'd recommended just creating a personal dev requirements.txt in your workspace like `requirements-mingshi.txt` that imports `dev-requirements.txt` and adds these stubs, but wouldn't currently add it to `requirements.txt` (`pyarrow-stubs` looks like a personal project for now).

This may also go into creating better IDE-specific setup instructions for developers in `README-development.md`.

Reply:
Yes I’m using VSCode which doesn't seem to have good support for Cython files. I see using PyCharm automatically gets the type information even without the stubs. 

I didn’t realize that the `pyarrow-stubs` was a personal project. Have removed the dependency in the latest commit.